scripts/primarybeamimage-fits.py

```diff
@@ -1,33 +1,56 @@
+#!/usr/bin/env python3
 import argparse
-import os
 import logging
-import numpy as np
-import primary_beam_correction as pbc
+import os
+
+import katsdpimageutils.primary_beam_correction as pbc
+
+
+def initialize_logs():
+    """
+    Initialize the log settings
+    """
+    logging.basicConfig(format='%(message)s', level=logging.INFO)
+
+
+def create_parser():
+    parser = argparse.ArgumentParser("Input a MeerKAT SDP pipeline continuum image and produce "
+                                     "primary beam corrected image.")
+    parser.add_argument('input',
+                        help='MeerKAT continuum uncorrected primary beam fits file')
+    parser.add_argument('--dir',
+                        help="The output full path plus file name with '.fits' extention. "
+                        "e.g. /home/name/primary_beam_image.fits. Default: same directory as "
+                        "the input image.")
+    return parser
+
 
 def main():
     # Initializing the log settings
-    pbc.intialize_logs()
+    initialize_logs()
     logging.info('MeerKAT SDP continuum image primary beam correction.')
-    parser =  pbc.create_parser()
+    parser = create_parser()
     args = parser.parse_args()
     path = os.path.abspath(args.input)
+    outpath = args.dir
     logging.info('----------------------------------------')
-    logging.info('Getting the beam pattern for each frequecy plane based on the '
+    logging.info('Getting the beam pattern for each frequency plane based on the '
                  'Cosine-squared power approximation from Mauch et al. (2020).')
-    bp = pbc.beam_pattern(path)
+    bp, raw_image = pbc.beam_pattern(path)
     # pbc - primary beam corrected
     logging.info('----------------------------------------')
-    #logging.info('Doing the primary beam correction in each frequency plane and averaging')
-    pbc_image = pbc.primary_beam_correction(bp, path)
-    # Saving the primary beam corrected image
+    logging.info('Doing the primary beam correction in each frequency plane and averaging')
+    pbc_image = pbc.primary_beam_correction(bp, raw_image, px_cut=0.1)
     logging.info('----------------------------------------')
     logging.info('Saving the primary beam corrected image')
-    head, tail = os.path.split(path)
-    fname, fext = tail.split('.')
-    outputpath = (head+fname+'_PB'+fext)
+    fpath, fext = os.path.splitext(path)
+    if outpath is None:
+        outputpath = fpath+'_PB' + fext
+    else:
+        outputpath = outpath
```

Comment:
`outputpath` -> `output_path` would be easier to read and consistent with Python conventions.

Reply:
Corrected.